test/support/admin_edition_controller_test_helpers.rb

```diff
@@ -753,32 +753,6 @@ def should_allow_role_appointments_for(edition_type)
       end
     end
 
-    def should_prevent_modification_of_unmodifiable(edition_type)
```

Comment:
The `prevent_modification_of_unmodifiable_edition` is still used by several (non-standalone Edition) controllers (such as Call for Evidence responses, or attachments) - but these don't appear to be tested. Could probably repurpose this deleted method to test those, but feels out of scope for this PR given I'm not making anything worse.